src/engine/source/indexerconnector/include/indexerConnector/indexerConnector.hpp

```diff
@@ -30,7 +28,32 @@
 #define EXPORTED
 #endif
 
-static constexpr auto DEFAULT_INTERVAL = 60000u;
+/**
+ * @brief Options for the secure communication.
+ *
+ */
+struct SslOptions
+{
+    std::vector<std::string> cacert;
+    std::string cert;
+    std::string key;
+};
+
+/**
+ * @brief Options for the IndexerConnector.
+ *
+ */
+struct IndexerConnectorOptions
+{
+    std::string name;
+    std::vector<std::string> hosts;
+    std::string username;
+    std::string password;
+    SslOptions sslOptions;
+    uint32_t timeout = 60000u;
+    uint8_t workingThreads = 1;
+    std::string databasePath;
+};
```

Comment:
it is feasible to consolidate both structures into one if SslOptions is not used in contexts other than IndexerConnectorOptions. Integration into a single structure can simplify code design and usage, especially if you do not anticipate that SslOptions will need to be reused or extended elsewhere. This approach also improves code cohesion by clearly grouping all related fields into a single logical entity.
On the other hand it is necessary to specify the documentation of the configuration in this structure that defines it and not in the constructor. Previously this was like this because it received a JSON and there was no place to document the schema.
```suggestion
/**
 * @brief Configuration options for the Indexer Connector.
 * 
 */
struct IndexerConnectorOptions
{
    std::string name;               ///< The name of the index to push the data.
    std::vector<std::string> hosts; ///< The list of hosts to connect to. i.e. ["https://localhost:9200"]
    std::string username;           ///< The username to authenticate with OpenSearch.
    std::string password;           ///< The password to authenticate with OpenSearch.

    struct
    {
        std::vector<std::string> cacert; ///< The list of CA certificates to trust.
        std::string cert;                ///< The certificate to connect to OpenSearch.
        std::string key;                 ///< The key to connect to OpenSearch.
    } sslOptions;                        ///< The SSL options to connect to OpenSearch.

    uint32_t timeout = 60000u;  ///< The timeout in milliseconds to connect to OpenSearch.
    uint8_t workingThreads = 1; ///< The number of threads to dequeue and send the data.
    std::string databasePath;   ///< The path to the database file.
};
```